kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterServiceDependentResource.java

```diff
@@ -72,28 +74,40 @@ public Map<String, Service> desiredResources(
                 .flatMap(ProxyNetworkingModel.ClusterNetworkingModel::requiredSniLoadbalancerPorts)
                 .distinct().sorted().toList();
 
-        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts);
+        Set<BootstrapServersAnnotation.BootstrapServer> bootstraps = clusterNetworkingModels.stream()
+                .flatMap(networking -> {
+                    return networking.clusterIngressNetworkingModelResults().stream()
+                            .map(ProxyNetworkingModel.ClusterIngressNetworkingModelResult::clusterIngressNetworkingModel)
+                            .filter(clusterIngressModel -> clusterIngressModel.requiredSniLoadBalancerServicePorts()
+                                    .findAny().isPresent())
+                            .map(clusterIngressModel -> new BootstrapServersAnnotation.BootstrapServer(ResourcesUtil.name(clusterIngressModel.cluster()),
+                                    ResourcesUtil.name(clusterIngressModel.ingress()), clusterIngressModel.bootstrapServers()));
+                })
+                .collect(Collectors.toSet());
+
+        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts, bootstraps);
 
         return Stream.concat(serviceStream, sniServiceStream).collect(toByNameMap());
     }
 
-    private ObjectMeta serviceMetadata(KafkaProxy primary, String name) {
+    private ObjectMeta serviceMetadata(KafkaProxy primary, String name, Set<BootstrapServersAnnotation.BootstrapServer> bootstraps) {
```

Comment:
Should the `bootstraps` nullable? I would have thought we don't want an annotation with the empty list on every service, but probably I don't quite understand what you're doing.

Reply:
In this case the LoadBalancer Service is only manfested if at least one ingress needs a LoadBalancer Service, so we should have at least 1 bootstrap.

Reply:
have also added code to `BootstrapServersAnnotation` so that empty set will not cause an annotation to be added (I changed it so we pass the meta fluent to `BootstrapServersAnnotation` and it decides what to write into it, if anything)

Reply:
method rename pushed up